pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -27,6 +30,10 @@ func (pp *ProxyProcessor) OnInvokeStart(startDetails *proxy.InvocationStartDetai
 	log.Debug("[proxy] Invocation invokeHeaders are :", startDetails.InvokeHeaders)
 	log.Debug("[proxy] Invocation invokeEvent payload is :", startDetails.InvokeEventPayload)
 	log.Debug("[proxy] ---------------------------------------")
+
+	if !pp.DetectLambdaLibrary() {
```

Comment:
are we sure that /hello will be hit at this time?

Reply:
We won't be sure at the start of the invocation, so it is possible we could start a span even though the library is present. However, by the end of the invocation, I think we are sure, because the library calls the hello route before the customer code is run.

Reply:
💯